### PR TITLE
Docs: Improve commit message template

### LINKED ISSUE #1844: [Docs] Improve commit message template
# 📚 Docs issue

The current commit template looks like this:

```
# [!] The commit message should have the following form:
#
# ---
# <Tag>: Capitalized, summary (less then 50 chars)
# <blank line>
# If necessary, more detailed explanatory text, wrapped to about 72
# characters or so.
#
# The commit message should be in the imperative: "Fix bug" and not
# "Fixed bug" or "Fixes bug."  This convention matches up with commit
# messages generated by commands like \`git merge\` and \`git revert\`.
#
# If applicable, at the end, write where the users can find more
# information (URL(s), or other related issues), and/or what this
# commit fixes.
#
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
#
# Ref <URL/issue/issue comment>
#
# Fix #<issue>
# Close #<issue>
# ---
#
# For more information related to commit messages, please see:
# https://webhint.io/docs/contributor-guide/contributing/pull-requests/#commitmessages
#`;
```

There are a few things that are outdated or that cause confussion:

- Title can be now longer than 50 characters
- The links section at the end is confussing and people put the same ID in Ref, Fix and Close. That's kind of what we are asking but now what we want... We probably only need `Fix` and in some cases `Ref` when we point to comments or other links for reference. We've been using `Close` to indicate what PR the commit closes but there's no way for the developer to know until it's done so we can probably drop it and manually add it when we squash via the GitHub interface if needed (or manually)

### PR DESCRIPTION
**Pull request by karansapolia**
## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/webhintio/hint)
- [x] Followed the [commit message guidelines](https://webhint.io/docs/contributor-guide/getting-started/pull-requests/#commit-messages)

For non-trivial changes, please make sure you also:

- ~~[ ] Added/Updated related documentation.~~
- ~~[ ] Added/Updated related tests.~~

## Short description of the change(s)
Improves existing commit message template.
Fixes #1844

### CODE REVIEW

**Comment by karansapolia** [comment]
Sure. Should i refer to `Ref <url/issue/issue comment>` in the prose or just a mention to referencing to related links in the last para?

**Comment by antross** [comment]
I was thinking of something near the end of that section like:
> To **refer** to an issue, comment, or URL in the description which is not being resolved by the commit, use the form `Ref <reference>`. For example:
> `Ref #1911`
> `Ref https://github.com/webhintio/hint/pull/1911#issuecomment-463715363`
> `Ref https://webhint.io/docs/contributor-guide/`

**Comment by karansapolia** [comment]
> If applicable, to refer to an issue, comment or URL not being resolved
by the commit or to link to more related information, use the form 
'Ref \<reference\>' below the seperator(-------) and before
'Fix #\<issue\>'.For example: `Ref #1911`,
`Ref https://webhint.io/docs/contributor-guide`,
`Ref https://github.com/webhintio/hint/pull/1911#issuecomment-463715363`

Will this do?

**Comment by antross** [comment]
Works for me. 👍

**Comment by molant** [comment]
@karansapolia looks like you have a [linting problem](https://dev.azure.com/webhint/webhint/_build/results?buildId=1748&view=logs&jobId=87b22c66-89ad-5aee-5190-c2f040f78e46&taskId=911263ff-bf20-50b2-0ed6-a32f919d9e58&lineStart=145&lineEnd=145&colStart=37&colEnd=44)

```
/home/vsts/work/1/s/scripts/prepare-commit-message.js
  11:31  error  Trailing spaces not allowed  no-trailing-spaces

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.
```

Can you please fix it?

**Comment by karansapolia** [comment]
Oh right! That was caused in effort to remove the first empty line in the commit message template because of the leading backtick. I'll keep it as it was before.

File: packages/hint/docs/contributor-guide/getting-started/pull-requests.md [hunk 1]
```diff
@@ -75,25 +75,21 @@ git commit
...
-"Fixed bug" or "Fixes bug."  This convention matches up with commit
-messages generated by commands like `git merge` and `git revert`.
+If necessary, more detailed explanatory text and/or what this commit
+fixes, wrapped to about 72 characters or so. The commit message should
+be in the imperative: "Fix bug" and not "Fixed bug" or "Fixes bug."
 
-If applicable, at the end, write where the users can find more
-information (URL(s), or other related issues), and/or what this
-commit fixes.
+If applicable, to refer to an issue, comment or URL not being resolved
```
Comment (antross) [inline packages/hint/docs/contributor-guide/getting-started/pull-requests.md]: Ah, I think I wasn't clear. I was thinking of putting the info about `Ref` in the text outside of the example after
> **All other lines**, where possible, should be wrapped to about 72 characters.

That way we keep the example in the documentation matching the commit message template.

Comment (karansapolia) [reply packages/hint/docs/contributor-guide/getting-started/pull-requests.md]: Do you mean that we include the info about `Ref` in the same para, maybe after
> where possible, should be wrapped to about 72 characters.

And have the examples

> For example:
`Ref #1911`
`Ref https://github.com/webhintio/hint/pull/1911#issuecomment-463715363`
`Ref https://webhint.io/docs/contributor-guide/`

at the end of this para?

> That way we keep the example in the documentation matching the commit message template.

My understanding is we are updating both the commit message template at [`prepare-commit-message.js`](https://github.com/webhintio/hint/blob/master/scripts/prepare-commit-message.js#L11) and the example at [`pull-request.md`](https://github.com/webhintio/hint/blob/master/packages/hint/docs/contributor-guide/getting-started/pull-requests.md#step-2-make-your-changes), to match the new simpler template we decided: 
```
const NEW_GIT_COMMIT_MESSAGE = `
### Please follow commit style. Before committing see:
### https://webhint.io/docs/contributor-guide/getting-started/pull-requests/#commit-messages
###
# <Tag>: Capitalized, summary (50 chars recommended)
#
# If necessary, more detailed explanatory text and/or what this commit
# fixes, wrapped to about 72 characters or so. The commit message should
# be in the imperative: "Fix bug" and not "Fixed bug" or "Fixes bug."
#
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
#
# Fix #<issue>
`;
```
Let me know where I miss your point.

Comment (antross) [reply packages/hint/docs/contributor-guide/getting-started/pull-requests.md]: > Do you mean that we include the info about `Ref` in the same para, maybe after
> > where possible, should be wrapped to about 72 characters.

You got it.

> My understanding is we are updating both the commit message template at [`prepare-commit-message.js`](https://github.com/webhintio/hint/blob/master/scripts/prepare-commit-message.js#L11) and the example at [`pull-request.md`](https://github.com/webhintio/hint/blob/master/packages/hint/docs/contributor-guide/getting-started/pull-requests.md#step-2-make-your-changes), to match the new simpler template we decided:

Yes, exactly.

Comment (karansapolia) [reply packages/hint/docs/contributor-guide/getting-started/pull-requests.md]: Okay, got it. You mean adding info about `Ref` in the paragraphs after the example in pull-requests.md. Specifically after 
> **All other lines**, where possible, should be wrapped to about 72 characters.
And not including it in the template, as I have done. Sorry, my bad. Will remove it from both the example and the `prepare-commit-message.js` template, and add it in the documentation below the example.

Comment (karansapolia) [reply packages/hint/docs/contributor-guide/getting-started/pull-requests.md]: Sorry, deleted the earlier comment. It felt too long, and confusing. This one captures it. Sorry quite rusty at working on Github.

**Comment by karansapolia** [comment]
@antross I hope the changes are all right now. Let me know.

**Comment by karansapolia** [comment]
Thank you all! :raised_hands: This is my second merged PR here. Feeling happy and learning a lot along the way!

**Comment by molant** [comment]
Thanks for your contributions!

**Pull request merged**